src/stacked_borrows.rs

```diff
@@ -900,11 +826,14 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // We have to use shared references to alloc/memory_extra here since
                 // `visit_freeze_sensitive` needs to access the global state.
                 let extra = this.get_alloc_extra(alloc_id)?;
+
                 let mut stacked_borrows = extra
                     .stacked_borrows
                     .as_ref()
                     .expect("we should have Stacked Borrows data")
                     .borrow_mut();
+                let mut current_span = this.machine.current_span();
```

Comment:
What changed that makes it so that we need a new `current_span` here, rather than reusing the old one as before?

Reply:
:thinking: me being bad at rebasing probably